.travis.yml

```diff
@@ -0,0 +1,35 @@
+---
+sudo: required
+
+env:
+  - distribution: ubuntu
+    version: xenial
+    ansible: 2.4.3.0
+  - distribution: ubuntu
+    version: xenial
+    ansible: 2.3.1.0
+  - distribution: ubuntu
+    version: trusty
+    ansible: 2.4.3.0
+  - distribution: ubuntu
+    version: trusty
+    ansible: 2.3.1.0
+    
+services:
+  - docker
+
+before_install:
```

Comment:
@thraxil Looks like a complicated setup, is this usual Travis/edX stuff?

Wondering if you could handle this _without_ using Docker?

 - [Multi-OS/distro with Travis](https://docs.travis-ci.com/user/multi-os/)

Reply:
I run this locally as well, and Docker is a much simpler setup for the matrix of distros/ansible versions than anything else I can come up with.

I also wanted to ensure that the environment is minimal. Travis' various OS environments come with a ton of extra packages already installed (because they presume that you will want to run unit tests of various types in them). For testing these ansible roles, we want them running against a system that is as barebones as possible. Hence Dockerfiles that install the bare minimum to get ansible to run.